Parser/tokenizer.c

```diff
@@ -1021,6 +1050,71 @@ tok_readline_raw(struct tok_state *tok)
     return 1;
 }
 
+static int
+tok_readline_string(struct tok_state* tok) {
+    PyObject* line = NULL;
+    PyObject* raw_line = PyObject_CallNoArgs(tok->readline);
+    if (raw_line == NULL) {
+        if (PyErr_ExceptionMatches(PyExc_StopIteration)) {
+            PyErr_Clear();
+            return 1;
+        }
+        error_ret(tok);
```

Comment:
Shouldn't the `if` for `PyMem_Free` in `error_ret` be changed as well (like in `_PyTokenizer_Free`)?